inspector/src/main/java/cern/jarrace/inspector/entry/EntryListenerFactory.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * © Copyright 2016 CERN. This software is distributed under the terms of the Apache License Version 2.0, copied
+ * verbatim in the file “COPYING”. In applying this licence, CERN does not waive the privileges and immunities granted
+ * to it by virtue of its status as an Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package cern.jarrace.inspector.entry;
+
+import com.sun.jdi.ThreadReference;
+
+/**
+ * A factory for creating new {@link EntryListener}s. The method {@link #onBreakpoint(ThreadReference, EntryState)}
+ * is called every time a new instance has been created and reached the first breakpoint defined in the class. This
+ * factory produces {@link EntryListener}s which will receive callback events from the running JDI whenever
+ * a new line is stepped over ({@link EntryListener#onLocationChange(EntryState)}) or whenever this
+ * instance is killed ({@link EntryListener#onInspectionEnd(EntryState)}).
+ *
+ * @param <Listener> The type of EntryListener to return.
+ * @author jepeders
+ */
+@FunctionalInterface
+public interface EntryListenerFactory<Listener extends EntryListener> {
```

Comment:
1) Usually, we keep to the convention that types are single letters (could be L in this case)
2) Why doe you need this extends clause anyhow? If I look at the method below, there is no way to constrain the Listener, so I doubt that there is a useful usage of the method other than 
`EntryListener listener = onNewEntry(...)`